event/event.go

```diff
@@ -239,11 +241,59 @@ type ReleaseEventCommon struct {
 	Error string `json:"error,omitempty"`
 }
 
+const (
+	// ReleaseImageSpecType is a type of release spec when there are update.Images
+	ReleaseImageSpecType = "releaseImageSpecType"
+	// ReleaseContainersSpecType is a type of release spec when there are update.Containers
+	ReleaseContainersSpecType = "releaseContainersSpecType"
+)
+
+// ReleaseSpec is a spec for images and containers release
+type ReleaseSpec struct {
+	// Type is ReleaseImageSpecType or ReleaseContainersSpecType
+	// if empty (for previous version), then use ReleaseImageSpecType
+	Type                  string
+	ReleaseImageSpec      *update.ReleaseImageSpec
+	ReleaseContainersSpec *update.ReleaseContainersSpec
+}
+
+// UnmarshalJSON for old version of spec (update.ReleaseImageSpec) where Type is empty
+func (s *ReleaseSpec) UnmarshalJSON(b []byte) error {
+	type T ReleaseSpec
+	t := (*T)(s)
+	if err := json.Unmarshal(b, &t); err != nil {
+		return err
+	}
+
+	switch t.Type {
+	case "":
+		var r update.ReleaseImageSpec
+		if err := json.Unmarshal(b, &r); err != nil {
+			return err
+		}
+		s.ReleaseImageSpec = &r
+		s.Type = ReleaseImageSpecType
+
+	case ReleaseImageSpecType, ReleaseContainersSpecType:
+		r := (*T)(s)
+		if err := json.Unmarshal(b, &r); err != nil {
```

Comment:
`r` is already a pointer; but moreover, this decoding has already been done, in line 264. I think this case is a no-op -- all the fields will have been decoded. (Is there a test that doesn't pass unless this line is present?)

Reply:
Fixed.